From 48aea96a33a455d7c2cca3f57bef1a23213fcab4 Mon Sep 17 00:00:00 2001
From: Evan Wang <xswang@marvell.com>
Date: Tue, 26 Sep 2017 16:13:23 +0800
Subject: [PATCH 2097/2241] dts: cp110: differentiate SAR with new compatible
 string

On a8k, there are CP-0 and CP-1, and their functions are
not completely same in sample at reset, so it is necessary
to differentiate them.
The patch differentiates CP0 SAR and CP1 SAR with different
compatible string.
For A7K, the string is "marvell,a70x0-sample-at-reset".
For A8K CP0, it is "marvell,a80x0-cp0-sample-at-reset".
For A8K Cp1, it is "marvell,a80x0-cp1-sample-at-reset".

Change-Id: I29cc45fc39b781254b06afa5a36a0e51f32f2595
Signed-off-by: Evan Wang <xswang@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/44671
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-7020.dtsi  | 3 +++
 arch/arm64/boot/dts/marvell/armada-7040.dtsi  | 3 +++
 arch/arm64/boot/dts/marvell/armada-8020.dtsi  | 6 ++++++
 arch/arm64/boot/dts/marvell/armada-8040.dtsi  | 6 ++++++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi | 1 -
 5 files changed, 18 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7020.dtsi b/arch/arm64/boot/dts/marvell/armada-7020.dtsi
index ff194e7..194aaa7 100644
--- a/arch/arm64/boot/dts/marvell/armada-7020.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7020.dtsi
@@ -177,6 +177,9 @@
 			axim-hb1-wr@3c9000 {
 				status = "disabled";
 			};
+			sar@400200 {
+				compatible = "marvell,a70x0-sample-at-reset";
+			};
 		};
 		pcie0@600000 {
 			msi-parent = <&gic_v2m0>;
diff --git a/arch/arm64/boot/dts/marvell/armada-7040.dtsi b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
index d9b4a6f..835a0b7 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
@@ -185,6 +185,9 @@
 			axim-hb1-wr@3c9000 {
 				status = "disabled";
 			};
+			sar@400200 {
+				compatible = "marvell,a70x0-sample-at-reset";
+			};
 		};
 		pcie0@600000 {
 			msi-parent = <&gic_v2m0>;
diff --git a/arch/arm64/boot/dts/marvell/armada-8020.dtsi b/arch/arm64/boot/dts/marvell/armada-8020.dtsi
index 3e5384a..ab99a69 100644
--- a/arch/arm64/boot/dts/marvell/armada-8020.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8020.dtsi
@@ -215,6 +215,9 @@
 			cp0_usb3h1: usb3@510000 {
 				#stream-id-cells = <1>;
 			};
+			sar@400200 {
+				compatible = "marvell,a80x0-cp0-sample-at-reset";
+			};
 		};
 		axim-hb1-rd@3c8000 {
 			status = "disabled";
@@ -267,6 +270,9 @@
 			rtc@284000 {
 				status = "okay";
 			};
+			sar@400200 {
+				compatible = "marvell,a80x0-cp1-sample-at-reset";
+			};
 		};
 		axim-hb1-rd@3c8000 {
 			status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-8040.dtsi b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
index 9fd80c6..f209afa 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
@@ -228,6 +228,9 @@
 			cp0_usb3h1: usb3@510000 {
 				#stream-id-cells = <1>;
 			};
+			sar@400200 {
+				compatible = "marvell,a80x0-cp0-sample-at-reset";
+			};
 		};
 		axim-hb1-rd@3c8000 {
 			status = "disabled";
@@ -280,6 +283,9 @@
 			rtc@284000 {
 				status = "okay";
 			};
+			sar@400200 {
+				compatible = "marvell,a80x0-cp1-sample-at-reset";
+			};
 		};
 		axim-hb1-rd@3c8000 {
 			status = "disabled";
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 49a00ec..fe7c9a5 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -544,7 +544,6 @@
 			};
 
 			CP110_LABEL(sar): sar@400200 {
-				compatible = "marvell,sample-at-reset-cp110";
 				reg = <0x400200 0x8>;
 				status = "okay";
 			};
-- 
2.7.4

